ts/screens/wallet/WalletsScreen.tsx

```diff
@@ -187,7 +193,26 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
     )
 });
 
+const mergeProps = (
```

Comment:
Instead of using mergeProps you should have all stuffs you need.
Could it be enough change this block

```typescript
<AddPaymentMethodButton
              onPress={this.props.navigateToWalletAddPaymentMethod}
            />
```

 with this?

```typescript
<AddPaymentMethodButton
              onPress={() =>
                this.props.navigateToAddPaymentMethod(
                  getCurrentRouteKey(this.props.nav)
                )
              }
            />
```

Reply:
yes, looks good don't know what brought me to the solution of the `mergeProps` at the end, maybe i was too focused on following the navigation flow to propagate the information